crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -216,6 +216,84 @@ impl<'a, T: Resource> FetchSystemParam<'a> for FetchResMut<T> {
     }
 }
 
+pub struct FetchOptionRes<T>(PhantomData<T>);
+
+impl<'a, T: Resource> SystemParam for Option<Res<'a, T>> {
+    type Fetch = FetchOptionRes<T>;
+}
+
+impl<'a, T: Resource> FetchSystemParam<'a> for FetchOptionRes<T> {
+    type Item = Option<Res<'a, T>>;
+
+    fn init(system_state: &mut SystemState, _world: &World, _resources: &mut Resources) {
+        if system_state.resource_access.is_write(&TypeId::of::<T>()) {
+            panic!(
+                "System `{}` has a `Res<{res}>` parameter that conflicts with \
+                another parameter with mutable access to the same `{res}` resource.",
+                system_state.name,
+                res = std::any::type_name::<T>()
+            );
+        }
+        system_state.resource_access.add_read(TypeId::of::<T>());
+    }
+
+    #[inline]
+    unsafe fn get_param(
+        _system_state: &'a SystemState,
+        _world: &'a World,
+        resources: &'a Resources,
+    ) -> Option<Self::Item> {
+        if resources.contains::<T>() {
+            Some(Some(Res::new(
+                resources.get_unsafe_ref::<T>(ResourceIndex::Global),
+            )))
+        } else {
+            Some(None)
+        }
```

Comment:
I'm not sure if this is cleaner or not, but it's worth considering
```suggestion
        Some(
            resources
                .contains::<T>()
                .then(|| Res::new(resources.get_unsafe_ref::<T>(ResourceIndex::Global))),
        )
```

Reply:
I prefer the imperative style over the functional style.

So i think the original is clearer and more readable.